terraform_apply.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# A script for creating assessment environments using
+# cisagov/cool-assessment-terraform.
+#
+# This is necessary because the CloudWatch alarm resources associated
+# with the EC2 instances are created using for_each expressions that
+# are dynamic-ish.
+#
+# Examples:
+# - See what would be created:
+#   $ AWS_PROFILE=cool-user AWS_SHARED_CREDENTIALS_FILE=~/.aws/production_credentials AWS_DEFAULT_REGION=us-east-1 ./terraform_apply.sh -var-file=envX-production.tfvars
+# - Create it!  (You will not be prompted.)
+#   $ AWS_PROFILE=cool-user AWS_SHARED_CREDENTIALS_FILE=~/.aws/production_credentials AWS_DEFAULT_REGION=us-east-1 ./terraform_apply.sh -auto-apply -var-file=envX-production.tfvars
+
+# Export some environment variables that we want the terraform child
+# processes to inherit.
+export AWS_PROFILE
+export AWS_SHARED_CREDENTIALS_FILE
+export AWS_DEFAULT_REGION
+
+# Perform a targeted apply to create the EC2 instances, then create
+# everything else.  This is a workaround for the dynamic-ish for_each
+# expressions mentioned above.
+terraform apply "${@}" \
+  -target=aws_iam_role_policy_attachment.provisionassessment_policy_attachment \
```

Comment:
I always thought that we needed to first create the policy (`aws_iam_policy.provisionassessment_policy`) as well as the policy attachment.  If you only target the policy attachment, does that automatically target the policy too since it's a dependency?

Reply:
That definitely appeared to be what happened when I tested this code, and it makes sense to me that it would be the case.